Remotion/Validation/Core/Implementation/GenericTypeAwareValidatedTypeResolverDecorator.cs

```diff
@@ -46,9 +49,20 @@ public IValidatedTypeResolver InnerResolver
     {
       ArgumentUtility.CheckNotNull ("collectorType", collectorType);
 
-      if (TypeExtensions.CanAscribeTo (collectorType, typeof(IValidationRuleCollector<>)))
-        return TypeExtensions.GetAscribedGenericArguments (collectorType, typeof(IValidationRuleCollector<>)).Single();
-      return _validatedTypeResolver.GetValidatedType (collectorType);
+      var itemType = s_genericValidationRuleCollectorTypeCache.GetOrAdd (
+              collectorType,
+              static t =>
+              {
+                var canAscribeTo = typeof (IValidationRuleCollector).IsAssignableFrom (t) && t.CanAscribeTo (typeof (IValidationRuleCollector<>));
+                return ValueTuple.Create (
+                    canAscribeTo,
+                    canAscribeTo
+                        ? t.GetAscribedGenericArguments (typeof (IValidationRuleCollector<>))[0]
+                        : null);
+              })
+          .ItemType;
+
+      return itemType ?? _validatedTypeResolver.GetValidatedType (collectorType);
```

Comment:
please split
```suggestion
      if (itemType != null)
        return itemType 

      return _validatedTypeResolver.GetValidatedType (collectorType);
```

This provides better results during coverage analysis